### DISCUSSION TITLE
Initialize a class

### DISCUSSION THREAD

**Discussion by FdHerrera** [mfussenegger/nvim-jdtls#456] [Q&A]
Hello

I don't know if this feature is already there or not, but I been looking for it on issues and discussions with no luck.

What I want is some way to initialize a class with that boilerplate code you have to write in a new java file.

I mean, start a java file where, given a file SomeClass.java in directory com/example/java, generates something like:

```
package com.example.java;

public class SomeClass {

}
```

I have NvimTree, so when creating a new file, it is empty, and didn't find nothing helpful with `:help jdtls`

**Comment by mfussenegger** [comment]
eclipse.jdt.ls returns a `class` snippet that can do that as part of the completion responses.

You'll need a completion plugin like nvim-cmp or nvim-lsp-compl and a snippet plugin to use it.


https://user-images.githubusercontent.com/38700/227700779-672a7bbf-ca98-4b6d-8219-60dfdac683ef.mp4

**Reply by OnlyWick** [reply]
Is there a way to trigger a command after creating a file so that we can choose what content to generate? Using snippets requires filling in a lot of information.

In addition to automatically generating package information in code actions, is it possible to add some other commands as well?
![image](https://github.com/user-attachments/assets/6915c3d1-c1d1-431f-99f5-8f05f0eea0b3)
Perhaps it can be done similar to IntelliJ IDEA, giving us a selection?
![image](https://github.com/user-attachments/assets/3803b5a8-aca7-4a89-a7eb-43ebb0597443)

**Reply by brianespinoza** [reply]
Also interested in this. From what I can tell, we aren't able to choose a destination either.. Code actions will simply generate a sibling class. Would a prompt or subsequent code action to specify a target location be in scope of nvim-jdtls to handle?

<img width="485" alt="Screenshot 2025-04-24 at 2 31 47 PM" src="https://github.com/user-attachments/assets/4ef2288a-227a-453d-8596-670241088a7a" />
